numpy/core/src/multiarray/buffer.c

```diff
@@ -544,7 +547,6 @@ _buffer_info_new(PyObject *obj, int flags)
                 }
             }
         }
-        Py_INCREF(descr);
     }
 
     /* Fill in format */
```

Comment:
`descr` is not used until line 554. Could you move the `descr = PyArray_DESCR(arr); Py_INCREF(descr);` to there instead, or comment why it needs to stay at line 489?

Reply:
Yeah, much better, thanks, definitely moved the wrong line around (moved it above now as well, so its at the end of the block in both branches).